test/plugin_functional/test_suites/core_plugins/applications.ts

```diff
@@ -29,15 +29,22 @@ export default function ({ getService, getPageObjects }: PluginFunctionalProvide
     return (await wrapper.getSize()).height;
   };
 
-  const getKibanaUrl = (pathname?: string, search?: string) =>
-    url.format({
+  const getKibanaUrl = (pathname?: string, search?: string) => {
+    return url.format({
       protocol: 'http:',
       hostname: process.env.TEST_KIBANA_HOST || 'localhost',
       port: process.env.TEST_KIBANA_PORT || '5620',
       pathname,
       search,
     });
+  };
 
+  const waitForUrlToBeWithTimeout = async (pathname?: string, search?: string, time?: number) => {
+    const expectedUrl = getKibanaUrl(pathname, search);
+    return await retry.waitForWithTimeout('navigates to app root', time ?? 3000, async () => {
+      return (await browser.getCurrentUrl()) === expectedUrl;
+    });
+  };
   /** Use retry logic to make URL assertions less flaky */
   const waitForUrlToBe = (pathname?: string, search?: string) => {
```

Comment:
The existing method relies on `retry.waitFor`.
I've been digging into the code, and found out that the `waitFor` [implementation](https://github.com/elastic/kibana/blob/main/packages/kbn-ftr-common-functional-services/services/retry/retry.ts#L60) calls the `waitForWithTimeout` too.
It does so using `timeouts.waitFor` timeout config, which defaults to [20 seconds](https://github.com/elastic/kibana/blob/main/packages/kbn-test/src/functional_test_runner/lib/config/schema.ts#L117).

It's too bad that the CI artifacts collected upon failure can't show the browser URL bar, but I think that if this URL hasn't changed after 20 seconds, it likely won't change, so the problem must be somewhere else.

Reply:
Thanks for the tips, I'll look into it

Reply:
@gsoldevila it looks like most of the ftr tests not in x-pack that implement `appsMenu.clickLink(title)` directly are skipped, probably because they're failing.

The Dashboard tests aren't skipped though, so I implemented something similar.

Problem is, we still don't know what causes these failures and need to figure that out.